### PR TITLE
Add e2e-test guide

### PR DESCRIPTION
**Pull request by noobwei**

### CODE REVIEW

File: content/en/docs/developer/e2e-guide.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+draft: false
+linktitle: E2E test guide
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:3]:
```suggestion
linktitle: E2E Test Guide
```

File: content/en/docs/developer/e2e-guide.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+---
+End-to-End (E2E) testing is a crucial component in modern software development, designed to simulate user interactions across the entire application to ensure all components and integrations function together seamlessly. By incorporating E2E testing, we can verify that changes in code do not disrupt existing functionalities, maintaining the integrity and reliability of the system even as it evolves. 
+
+## Prerequisites
+
+The Go version in go.mod is 1.22.0
+
+- [Update the go version](https://go.dev/doc/install)
+- Use `go mod tidy`
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:20]: I think it's unnecessary to tell user to uprade go version here and why run `go mod tidy`?

Just tell user to install go 1.22 is necessary.

File: content/en/docs/developer/e2e-guide.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+The following components will be installed while using the shell
+
+- Kind
+- Helm
+- istioctl 
+
+Components that need to be installed by the user
+
+- [Docker](https://docs.docker.com/engine/install/)
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:30]: Put the components that user need to install at first.

Also we don't need to provide installation link, as they may change and we don't want to maintain them.

File: content/en/docs/developer/e2e-guide.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+- [kubectl](https://kubernetes.io/docs/tasks/tools/)
+
+## Usage
+
+To run the E2E tests, execute the `run_test.sh` script located in the `./test/e2e` directory. This script automates the following tasks:
+
+1. **Installing dependencies:** Tools like Kind, Helm, and Istioctl are installed.
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:37]:
```suggestion
1. **Installing dependencies:** Tools like Kind, Helm and Istioctl ...
```

File: content/en/docs/developer/e2e-guide.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+- **Full Test Run (First Time):**
+
+  ```bash
+  ./test/e2e/run_test.sh
+  ```
+
+  Use this command for the initial setup and test run to ensure everything is configured correctly.
+
+- **Subsequent Test Runs (Skipping Setup):**
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:62]:
```suggestion
- **Subsequent Test Runs (Skipping all setup and only run tests):**
```

File: content/en/docs/developer/e2e-guide.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+title: Run E2E test
+toc: true
+type: docs
+
+---
+End-to-End (E2E) testing is a crucial component in modern software development, designed to simulate user interactions across the entire application to ensure all components and integrations function together seamlessly. By incorporating E2E testing, we can verify that changes in code do not disrupt existing functionalities, maintaining the integrity and reliability of the system even as it evolves. 
+
+## Prerequisites
+
+The Go version in go.mod is 1.22.0, it is necessary to install go 1.22.
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:17]: Just add `- Go` before `- Docker` below. They would update their version if it's not appropriate.

File: content/en/docs/developer/e2e-guide.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+## Command Line Flags
+
+When testing locally, you may want to skip some setup steps to save time, especially after the initial setup is complete. The following flags are available to customize the test execution:
+
+- `--skip-install-dep`: Skips the installation of dependencies.
+- `--skip-build`: Skips building and pushing the Kmesh image to the local image registry.
+- `--skip-setup`: Skips deploying the Kubernetes cluster, Istio, and Kmesh.
+- `--only-run-tests`: Skips all other steps and focuses only on deploying test applications and running E2E tests.
+
+### **Example Commands**
```
Comment (YaoZengzeng) [inline content/en/docs/developer/e2e-guide.md:49]: Don't bold the title, it looks awkward

**Comment by hzxuzhonghu** [comment]
@YaoZengzeng Do we support specifying a case to run

And also do we support no cleanup?

File: content/en/docs/developer/e2e-guide.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+5. **Deploying test applications and executing the E2E tests.**
+
+## Command Line Flags
+
+When testing locally, you may want to skip some setup steps to save time, especially after the initial setup is complete. The following flags are available to customize the test execution:
+
+- `--skip-install-dep`: Skips the installation of dependencies.
+- `--skip-build`: Skips building and pushing the Kmesh image to the local image registry.
+- `--skip-setup`: Skips deploying the Kubernetes cluster, Istio, and Kmesh.
+- `--only-run-tests`: Skips all other steps and focuses only on deploying test applications and running E2E tests.
```
Comment (hzxuzhonghu) [inline content/en/docs/developer/e2e-guide.md:47]: Is there any other istio flags supported?

Comment (YaoZengzeng) [reply content/en/docs/developer/e2e-guide.md:47]: Now these are all the flags

**Comment by YaoZengzeng** [comment]
> @YaoZengzeng Do we support specifying a case to run
> 
> And also do we support no cleanup?

Currently it is not support a case to run and by default infra like k8s, istio and Kmesh will not cleanup, but test applications will.

**Comment by hzxuzhonghu** [comment]
Istio has some startup params, `istio.test.nocleanup` by setting to true, it will not cleanup applications. What about kmesh?

FYI https://github.com/istio/istio/wiki/Istio-Test-Framework#flags

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu maybe this PR could get merged first? And then continuously improved as the E2E test function is enhanced.

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**